language/move-core/types/src/value.rs

```diff
@@ -69,6 +69,11 @@ impl MoveFieldLayout {
 pub enum MoveStructLayout {
     /// The representation used by the MoveVM
     Runtime(Vec<MoveTypeLayout>),
+    /// The representation used by the MoveVM with extra unique identifier for distinct type.
+    CheckedRuntime {
+        fields: Vec<MoveTypeLayout>,
+        tag: u64,
```

Comment:
A comment explaining what this `u64` means would be useful. A descriptively named newtype would be even better.